### DISCUSSION TITLE
error on timeout - 200 and an error after timing out what gives?

### DISCUSSION THREAD

**Discussion by dugaquilin** [node-fetch/node-fetch#1495] [Q&A]
Hi, 

I'm new to node.js and even more to this module.  Big fan, like the work you have done.

I'm doing an HTTP GET and it returns a 200 after a redirect, which is fine, but then it also returns an error after timing out.

The web site is working, I don't understand why I'm getting an error.  
I was thinking of extending fetch, but it seems I can't export fetch as it is not a function.  

Any insights?

**Comment by jimmywarting** [comment]
do you have any sample code that you can show us what you are doing?

**Reply by dugaquilin** [reply]
Here is my code

```js
//-----------------------------------------------------------------------------
// Check all addresses.
// HTTP lookup - asynchronous
//-----------------------------------------------------------------------------
//console.log(await checkAllHTTP(myURLS,myDNSstatus));
async function checkAllHTTP(UrlList, mDNSs) {
  var funName = "async checkAllHTTP: "
  var i = 0;
  var HTTPstatus = [];
  var myStatus = "";

  console.log(funName + "=========================")
  console.log(funName + "... DNS  statusx " + mDNSs);
  console.log(funName + "... DNS  statusx " + mDNSs[i]);
  console.log(funName + "... DNS  UrlList " + UrlList);

  for (var i = 0; i < 20; i++) {
    console.log(funName + "... DNS  UrlListx " +UrlList[i] + "*")
  } //for loop - over rows

  i = 0;

  for (addy of UrlList) {
    console.log(funName + i + " ^" + mDNSs[i] + "^ ... DNS  status " + addy);

    if (mDNSs[i] === "fail") {
    //   console.log(funName + i + " ... HTTP skipping " + getHostname(addy))
    } else {
      //let myStatus = await myHTTPasy(addy)
      getData(addy);
      console.log(funName + i + " " + mDNSs[i] + " ... HTTP status  " +  myStatus + " " + getHostname(addy))
    }
    // console.log(funName + "...b- " + myStatus + " " + getHostname(addy));
    i += 1;
  }

  // Show that the array's are populated
  console.log(funName + "...DNS Status- "+ mDNSs[1]);
  console.log(funName + "...URL List  - "+ UrlList[1]);
  return mDNSs;
}

// ======================================================================
// ======================================================================
// ======================================================================



const getData = async url => {
  try {
    const response = await fetch(url);
    //const response = await fetchWrapper(url);
    //const json = await response.json();
    console.log("response:  ok " + response.redirected + " " + response.status + " " + response.url);
  } catch (error) {
    //console.log(error);
    console.log("response: err " + response.redirected + " " + response.status + " " + response.statusText);
    console.log("response: err getData: error");
  }
};

getData(url);
```

**Reply by dugaquilin** [reply]
this is a redacted output from the console.

response:  ok true 200 https://mysite.com
response: err true 200 OK
response: err getData: error

the last 2 items are from the same FQDN... the code pauses for a while at the 200 OK.. then puts out the getData: error message

**Reply by jimmywarting** [reply]
Hmm, without actually testing your code it looks to me that you aren't dealing with the response
it's always best to consume the response or abort the hole request if you no longer need it

so eg: 
```js
const getData = async url => {
  const response = await fetch(url);
  await response.arrayBuffer() // consumes the body and dose nothing with it
}
```
or: 
```js

const getData = async url => {
  const ctrl = new AbortController()
  const response = await fetch(url, { signal: ctrl.signal })
  ctrl.abort()
}
```

It could also be recommended to try out using http method `HEAD`, this will tell the server to just perform a get request but you aren't actually intrested in the body, only the response headers

**Reply by dugaquilin** [reply]
Oh, i never knew about the HEAD vs GET.  Thanks.

**Reply by dugaquilin** [reply]
What I don't understand is, I'm doing the catch(error) and it seems to be terminating the calling loop.  Since my loop is iterating over URLs I'd like to continue with the loop even though that one specific URL is broken.

Thank you for your code example.  It is advanced to me, it will take me a while to understand it.... Again.. I'm old school programmer new to node.js.

**Reply by dugaquilin** [reply]
I found that the page i am probing is actually not responding.
anyways... I am looking to capture the issue, and continue on. that is gracefully register the issue and continue on, but it is stopping.

I am probing multiple web sites and want to capture and continue but it is stuck.

FetchError: request to https://mysite.com/ failed, reason: connect ETIMEDOUT {ip address}

I'd like to tread the ETIMOUT as a trappable error code vs a crash.  For example, capture the ETIMEOUT  (because the site is non-responsive) then assign a status to it, then move on to the next url.

**Reply by dugaquilin** [reply]
My specific scenario is this.
1. the DNS entry returns an IP, so the site is "registered"
2. the FQDN is non-responsive - probably because the site is turned off and is non-responsive, so either the site was depricated, or the site is down, in which case I want to flag it as an issue and move on to probing the next site.  to me the lack of response is not the end of the world, it is just one of the expected outcomes
